pkg/util/util.go

```diff
@@ -129,3 +133,85 @@ func Int32PtrDerefOr(ptr *int32, def int32) int32 {
 	}
 	return def
 }
+
+// RemoveAllOneFilesystem removes path and any children it contains.
+// It removes everything it can but returns the first error
+// it encounters. If the path does not exist, RemoveAll
+// returns nil (no error).
+// It makes sure it does not cross mount boundary, i.e. it does *not* remove
+// files from another filesystems. Like 'rm -rf --one-file-system'.
+// It is copied from RemoveAll() sources, with IsLikelyNotMountPoint
+func RemoveAllOneFilesystem(mounter mount.Interface, path string) error {
+	// Simple case: if Remove works, we're done.
```

Comment:
os.Remove() will only remove an empty directory. So it shouldn't work if this is a mountpoint or if it has files in in.